test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URLDecoderTest.scala

```diff
@@ -0,0 +1,84 @@
+package org.scalajs.testsuite.javalib.net
+
+import org.junit.Test
+import org.junit.Assert._
+
+import java.net.URLDecoder
+import java.io.UnsupportedEncodingException
+
+class URLDecoderTest {
+
+  private final val utf8 = "utf-8"
+  private final val ReplacementChar = '\uFFFD'
+
+  @Test
+  def decodeTest(): Unit = {
+    @inline
+    def test(encoded: String, expected: String, enc: String = utf8): Unit =
+      assertEquals(URLDecoder.decode(encoded, enc), expected)
+
+    @inline
+    def illegalArgumentOrReplacement(encoded: String, enc: String = utf8): Unit = {
+      val thrown =
+        try {
+          val res = URLDecoder.decode(encoded, enc)
+
+          /* it is valid to return the Unicode replacement character (U+FFFD)
+            when encountering an invalid codepoint */
+          res.contains(ReplacementChar)
+        } catch {
+          case _: IllegalArgumentException => true
+        }
+
+      assertTrue(thrown)
+    }
+
+    @inline
+    def unsupportedEncoding(encoded: String, enc: String = utf8): Unit = {
+      val thrown =
```

Comment:
Should have braces `{..}` for the body on the `val thrown = { ...` because it has more than two lines.